### DISCUSSION TITLE
Image uploads - autogenerate thumbnail(s)?

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#1703] [Q&A] [answered]
When using Picture/Cropper/Upload I wish to be able to save attachments of BOTH the original uploaded file as well as autogenerated thumbnail(s). Is there some way to do this, if not via Orchid then through the Screen/Resource?

Example 1: for Bootstrap mobile layouts I wish to have small/medium/large variants of an image, so multiple thumbnails as separate files handled automatically. The thumbnail dimensions would be specified in the screen/resource ideally.

Example 2: there often are situations I wish to trigger REGENERATION of all thumbnails. Say for all "small" thumbnails I want to change them from 100px width to 120px width. For that you would need the original uploaded file. (I understand that something like this is very likely not an option in Orchid currently)

If anyone has any insight as to how to achieve something along those lines it would be appreciated. Thank you.

**Comment by tabuna** [comment, marked as answer]
Image requirements may vary. In addition, you do not always need to have several copies of the same image on your hard disk all the time. Try doing them dynamically, for example, https://github.com/Intervention/imagecache

You can define a static route and required formats. Then their creation and modification will be more effective:

https://github.com/Intervention/imagecache/blob/e714f13298ecaf9b2d11cb7106a0415d5615cbe5/src/config/config.php#L5-L71
